source/agora/crypto/Key.d

```diff
@@ -121,35 +147,51 @@ public struct PublicKey
     /*private*/ Point data;
     alias data this;
 
+    string hrp = HumanReadablePart;
+
     /// Construct an instance from binary data
-    public this (const Point args) pure nothrow @safe @nogc
+    public this (const Point args, const string hrp = HumanReadablePart)
+        nothrow @safe @nogc
     {
         this.data = args;
+
+        auto wkdata = WKKeyData.Keys[this.data];
+        if (wkdata != KeyData.init)
+            this.hrp = wkdata.hrp;
+        else
+            this.hrp = hrp;
     }
 
     /// Ditto
-    public this (const ubyte[] args) pure nothrow @safe @nogc
+    public this (const ubyte[] args, const string hrp = HumanReadablePart)
+        nothrow @safe @nogc
     {
         this.data = args;
+
+        auto wkdata = WKKeyData.Keys[this.data];
+        if (wkdata != KeyData.init)
+            this.hrp = wkdata.hrp;
+        else
+            this.hrp = hrp;
     }
 
     /// Uses Stellar's representation instead of hex
     public string toString () const @trusted nothrow
     {
-        ubyte[VersionWidth + PublicKey.sizeof] bin;
+        ubyte[VersionWidth + PublicKey.data.sizeof] bin;
```

Comment:
Hum that means that `PublicKey` is not a value type anymore, that's a big downside

Reply:
It's reverted.